### DISCUSSION TITLE
Rugged ConfigError safe.directory

### DISCUSSION THREAD

**Discussion by palash-07** [github-linguist/linguist#6104] [Q&A] [answered]
Linguist file outputs some weird error
```
 /usr/local/rvm/rubies/ruby-3.1.2/lib/ruby/gems/3.1.0/gems/github-linguist-7.12.2/bin/github-linguist:27:in `new': config value 'safe.directory' was not found (Rugged::ConfigError)
        from /usr/local/rvm/rubies/ruby-3.1.2/lib/ruby/gems/3.1.0/gems/github-linguist-7.12.2/bin/github-linguist:27:in `<top (required)>'
        from /usr/local/rvm/rubies/ruby-3.1.2/bin/github-linguist:25:in `load'
        from /usr/local/rvm/rubies/ruby-3.1.2/bin/github-linguist:25:in `<main>'
        from /usr/local/rvm/rubies/ruby-3.1.2/bin//ruby_executable_hooks:22:in `eval'
        from /usr/local/rvm/rubies/ruby-3.1.2/bin//ruby_executable_hooks:22:in `<main>'
]
```
Scan results inside the docker container

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x723873]
```

**Comment by lildude** [comment]
We're going to need a bit more information about your environment and commands you are running that are resulting in these errors.

**Reply by palash-07** [reply]
Firstly, I'm trying to build a docker image from a dockerfile which executes the following command 
`RUN gem install github-linguist -v 7.12.2`
After going into the repository folder inside the container, simply trying 
`<<SOME_PATH_IN_CONTAINER/github-linguist>> --json`

Sorry, I cannot fill you in with much details because this is part of a highly complex source code and its restricted.
Can you ask me specific questions about environment

**Reply by lildude** [reply]
Can you reproduce the errors using the linguist repo and Dockerfile exactly as documented [here](https://github.com/github/linguist/blob/master/README.md#docker)?

**Reply by palash-07** [reply]
It is working correctly on my local mac system but inside the docker container, when im trying to do the same thing, its failing.

I think the problem is based on mainly this 
`new': config value 'safe.directory' was not found (Rugged::ConfigError)`
which is associated with one of the lines of github-linguist script:
`repo = Rugged::Repository.new('.')`

I am certain this might be concerned with my docker environment but please help me to figure out what might have been wrong.

**Reply by palash-07** [reply]
without --json 
it shows the same parsing error with the following statement
`new': repository path '/tmp/source/slaps-scanner-test-java/' is not owned by current user (Rugged::ConfigError)`

**Reply by palash-07** [reply, marked as answer]
Sorry to bother you @lildude 
it was some folder's permission issue which is why I was facing this parsing error.
thanks for the support. You can close this discussion
